torch/_dynamo/variables/builder.py

```diff
@@ -1516,6 +1516,27 @@ def mark_static_input(self, value: torch.Tensor, guard: bool):
     def wrap_module(self, value: torch.nn.Module):
         from ..eval_frame import OptimizedModule
 
+        # if config.profile_params:
+        metrics_context = get_metrics_context()
+        if metrics_context.in_progress():
+            try:
+                params = (
+                    p
+                    for _, p in value.named_parameters(
+                        recurse=False, remove_duplicate=True
+                    )
+                    if not torch.nn.parameter.is_lazy(p)
+                )
+                metrics_context = get_metrics_context()
+                for p in params:
+                    metrics_context.set_key_value("param_numel_addr", id(p), p.numel())
+                    metrics_context.set_key_value("param_bytes_addr", id(p), p.nbytes)
+                    metrics_context.set_key_value("param_count_addr", id(p), 1)
```

Comment:
Change generator expression `params = (...)` to list comprehension `params = [...]`. Then move this out of the `try`-clause.

Reply:
Done.